test/filename-case.js

```diff
@@ -92,7 +92,89 @@ ruleTester.run('filename-case', rule, {
 		testManyCases('src/foo/___foo_bar.js', {snakeCase: true, pascalCase: true}),
 		testCaseWithOptions('src/foo/bar.js'),
 		testCase('src/foo/[fooBar].js', 'camelCase'),
-		testCase('src/foo/{foo_bar}.js', 'snakeCase')
+		testCase('src/foo/{foo_bar}.js', 'snakeCase'),
+		testCaseWithOptions(undefined, undefined, [
+			{case: 'kebabCase', ignore: ['FOOBAR\\.js']}
```

Comment:
It should allow specifying regex literals. Most people use JS configs, so it's nicer to also be able to specify literals.

Reply:
@sindresorhus just allow use `/foobar\.js/` as regexp, right? When i use string notation `foobar\\.js` it is also regexp?

Reply:
Allow both both regex literal and reges in a string.

Reply:
is `regex in a string` a good idea? maybe confusing people